internal/test/wmcb/pkginfo.go

```diff
@@ -0,0 +1,144 @@
+	sha string
+	// shaType is the type of SHA used, example: 256 or 512
+	shaType string
+}
+
+// pkgInfoFactory returns PkgInfo specific to the package name
+func pkgInfoFactory(name pkgName, shaType string, baseUrl string, version string) (PkgInfo, error) {
+	switch name {
+	case cniPluginPkgName:
+		return newCniPluginPkg(name, shaType, baseUrl, version)
+	case hybridOverlayPkgName:
+		return newHybridOverlayPkg(name, shaType)
+	default:
+		return nil, fmt.Errorf("invalid Package name")
+	}
+}
+
+// newCniPluginPkg returns cniPlugins implementation of PkgInfo interface
+func newCniPluginPkg(name pkgName, shaType string, baseUrl string, version string) (PkgInfo, error) {
+	if version == "" {
+		return nil, fmt.Errorf("latest cni plugins version not specified")
+	}
+	if baseUrl == "" {
+		return nil, fmt.Errorf("base url for cni plugins not specified")
+	}
+	return &cniPlugins{pkgInfo{
+		name:    name,
+		url:     baseUrl + version + "/cni-plugins-windows-amd64-" + version + ".tgz",
+		shaType: shaType}}, nil
+}
+
+// newHybridOverlayPkg returns hybridOverlay implementation of PkgInfo interface
+func newHybridOverlayPkg(name pkgName, shaType string) (PkgInfo, error) {
+	hybridOverlayUrl, err := framework.GetReleaseArtifactURL(hybridOverlayName)
+	if err != nil {
+		return nil, err
+	}
+	return &hybridOverlay{pkgInfo{
+		name:    name,
+		shaType: shaType,
+		url:     hybridOverlayUrl}}, nil
+}
+
+// getSHAFileContent returns the contents of the SHA file for the given url of the package.
+func (p *pkgInfo) getSHAFileContent() (string, error) {
+	pkgChecksumFileURL := p.url + "." + p.shaType
+	response, err := http.Get(pkgChecksumFileURL)
+	if err != nil {
+		return "", fmt.Errorf("could not fetch content from checksum file: %s", err)
+	}
+	defer response.Body.Close()
+
+	var checksumFileContent string
+	// Fetching checksum file content from the GET Response
+	scanner := bufio.NewScanner(response.Body)
+	for scanner.Scan() {
+		checksumFileContent = scanner.Text()
+	}
+	return checksumFileContent, nil
+}
+
+// getSHAValue returns the sha value of the cni plugins package
+func (c *cniPlugins) getSHAValue() (string, error) {
+	if c.sha != "" {
+		return c.sha, nil
+	}
+	checksumFileContent, err := c.getSHAFileContent()
+	if err != nil {
+		return "", err
+	}
+	// The checksum file content is in the format "<sha> <filename>". So to get SHA we need to extract only the <sha>
+	// from the file
+	sha512 := strings.Split(checksumFileContent, " ")
+	if len(sha512) < 2 {
+		return "", fmt.Errorf("checksum file content is not in the format : '<sha> <filename>'")
+	}
+	c.sha = sha512[0]
+	return c.sha, nil
+}
+
+// getPkgName returns the user defined name of the package
+func (p *pkgInfo) getPkgName() pkgName { return p.name }
+
+//getShaType returns the sha type of the package
+func (p *pkgInfo) getShaType() string { return p.shaType }
+
+//getPkgUrl returns the url of the package
+func (p *pkgInfo) getPkgUrl() string { return p.url }
```

Comment:
Please format for readability. `return ` should be on a new line. I know some codebases follow this, but I suggest we not go down that route.